### PR TITLE
feat: add user dialog and topbar docs

### PR DESCRIPTION
**Pull request by nkrantz**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: packages/paste-website/src/pages/components/topbar/index.mdx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+- [Account Switcher](/components/account-switcher)
+- Account type indicators, such as subaccount [Badges](/components/badge)
+- Possible upgrade actions
+
+User-specific controls are aligned to the right, such as:
+
+- Search [Input](/components/input)
+- Support [Button](/components/button)
+- [Product Switcher](/components/product-switcher)
+- [User Dialog](/components/user-dialog)
```
Comment (TheSisb) [inline packages/paste-website/src/pages/components/topbar/index.mdx:111]: thought: I don't think this info should be under the `Examples` heading, but above it like the previous version of this page.


```suggestion
Account-specific controls are aligned to the left, such as:

- [Account Switcher](/components/account-switcher)
- Account type indicators, such as subaccount [Badges](/components/badge)
- Possible upgrade actions

User-specific controls are aligned to the right, such as:

- Search [Input](/components/input)
- Support [Button](/components/button)
- [Product Switcher](/components/product-switcher)
- [User Dialog](/components/user-dialog)


## Examples
```

File: packages/paste-website/src/pages/components/topbar/index.mdx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+<LivePreview scope={{Topbar}} noInline>
+  {emptyTopbarExample}
```
Comment (TheSisb) [inline packages/paste-website/src/pages/components/topbar/index.mdx:114]: thought: The empty example looks buggy and underwhelming. Maybe just duplicate the one from the top of the page: defaultTopbarExample

Comment (serifluous) [reply packages/paste-website/src/pages/components/topbar/index.mdx:114]: +1

honestly, this whole section could just be removed. It doesn't add much more to the content.

File: packages/paste-website/src/pages/components/topbar/index.mdx (hunk 1 above)
Comment (serifluous) [reply packages/paste-website/src/pages/components/topbar/index.mdx:111]: The content could all be combined into one section, potentially into a table, like:

A Topbar is sticky to the top of the screen and consists of application-wide controls including:

| Account-specific controls aligned to the left                                                      | User-specific controls aligned to the right                |
|----------------------------------------------------------------------------------------------------|------------------------------------------------------------|
| - Account Switcher - Account type indicators, such as subaccount Badges - Possible upgrade actions | - Search - Support button - Product Switcher - User Dialog |

File: packages/paste-website/src/pages/components/topbar/index.mdx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+</LivePreview>
+
+## Usage guide
+
+### Figma
+
+Topbar, Account Switcher, Product Switcher, and User Dialog components are available in the Paste Components library as [Navigation components](https://www.figma.com/file/l2qtpQVMmXzQPVB3rEAA8i/%5BNEW%5D-Paste-Components?type=design&node-id=27837-187276&t=xcPacI1AN13MooNC-0).
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/topbar/index.mdx:121]:
```suggestion
Topbar is available in the Paste Components library as a [Navigation component](https://www.figma.com/file/l2qtpQVMmXzQPVB3rEAA8i/%5BNEW%5D-Paste-Components?type=design&node-id=27837-187276&t=xcPacI1AN13MooNC-0).
```

File: packages/paste-website/src/pages/components/user-dialog/index.mdx [hunk 4]
```diff
@@ -0,0 +1,235 @@
+export const meta = {
+  title: 'User Dialog',
+  package: '@twilio-paste/user-dialog',
+  description: 'User dialog is an expandable menu that contains profile related actions.',
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/user-dialog/index.mdx:4]:
```suggestion
  description: 'User Dialog is a menu that contains user profile-related actions.',
```

File: packages/paste-website/src/pages/components/user-dialog/index.mdx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+  {defaultUserDialogExample}
+</LivePreview>
+
+## Guidelines
+
+### About User Dialog
+
+A User Dialog is a composition of [Avatar](/components/avatar) and [Button](/components/button) that opens a dialog of items that trigger profile-related actions.
+
+A User Dialog item can be used to perform a single action, such as a page navigation or opening a modal. A profile-related action could be visiting a user settings page, switching the language of a product, or logging out.
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/user-dialog/index.mdx:93]:
```suggestion
A User Dialog is a composition of [Avatar](/components/avatar), [Button](/components/button), and the [Non-Modal Dialog primitive](/primitives/non-modal-dialog-primitive) that opens a dialog of profile-related actions. This includes visiting a user settings page, switching the language of a product, or logging out.
```

File: packages/paste-website/src/pages/components/topbar/index.mdx (hunk 2 above)
Comment (nkrantz) [reply packages/paste-website/src/pages/components/topbar/index.mdx:114]: I'll replace this example with the default example. Just a note that all of the examples are temporary anyways

File: packages/paste-website/src/pages/components/topbar/index.mdx (hunk 3 above)
Comment (nkrantz) [reply packages/paste-website/src/pages/components/topbar/index.mdx:121]: Thought it might be good to leave them in because theoretically if someone's coming to the topbar page, they'll also need some or all of these components. I will remove

Comment (serifluous) [reply packages/paste-website/src/pages/components/topbar/index.mdx:121]: Ya I think we should eventually group all the docs pages together so it's all easier to find.

File: packages/paste-website/src/pages/components/topbar/index.mdx (hunk 1 above)
Comment (nkrantz) [reply packages/paste-website/src/pages/components/topbar/index.mdx]: The table feels like squeezing too much new info together for me...

**Review by TheSisb — APPROVED** [review]

File: packages/paste-website/src/pages/components/topbar/index.mdx (hunk 1 above)
Comment (serifluous) [reply packages/paste-website/src/pages/components/topbar/index.mdx]: fair. The first list essentially repeats the 2nd and 3rd still tho. Hard to parse what the difference is between the three lists.

File: packages/paste-website/src/component-examples/TopbarExamples.ts [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+          <MenuItem {...menu} onClick={()=>onClick('Account three')}>
+            Account three
+          </MenuItem>
+        </Menu>
+        <Badge variant="subaccount" size="small">Subaccount</Badge>
+        <Box flex-shrink="4" paddingX="space150">
+          <Input placeholder="Search" insertBefore={<SearchIcon decorative/>}/>
+        </Box>
+        <ProductSwitcherButton {...productSwitcher} i18nButtonLabel="Switch products" />
+        <ProductSwitcher {...productSwitcher} aria-label="Avaiable accounts">
```
Comment (serifluous) [inline packages/paste-website/src/component-examples/TopbarExamples.ts:33]:
```suggestion
        <ProductSwitcher {...productSwitcher} aria-label="Available accounts">
```

File: packages/paste-website/src/component-examples/TopbarExamples.ts [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+`.trim();
+
+export const defaultProductSwitcherExample = `
+const ProductSwitcherMenu = () => {
+    const productSwitcher = useProductSwitcherState();
+    const [product, setProduct] = React.useState('twilio');
+    return (
+      <>
+        <ProductSwitcherButton {...productSwitcher} i18nButtonLabel="Switch products" />
+        <ProductSwitcher {...productSwitcher} aria-label="Avaiable accounts">
```
Comment (serifluous) [inline packages/paste-website/src/component-examples/TopbarExamples.ts:154]:
```suggestion
        <ProductSwitcher {...productSwitcher} aria-label="Available products">
```

File: packages/paste-website/src/component-examples/TopbarExamples.ts (hunk 6 above)
Comment (nkrantz) [reply packages/paste-website/src/component-examples/TopbarExamples.ts:33]: these were copied directly from the product switcher docs which were already merged, I'll update both

File: packages/paste-website/src/pages/components/topbar/index.mdx (hunk 1 above)
Comment (nkrantz) [reply packages/paste-website/src/pages/components/topbar/index.mdx]: ok, combined them into something I think is more clear

**Review by serifluous — DISMISSED** [review]

**Review by serifluous — APPROVED** [review]

**Review by SiTaggart — APPROVED** [review]
